ads/advertserve.js

```diff
@@ -21,7 +21,23 @@ import {validateData, writeScript} from '../3p/3p';
  * @param {!Object} data
  */
 export function advertserve(global, data) {
-  validateData(data, [], ['zid', 'pid', 'client']);
+  validateData(
+    data,
+    ['zid', 'pid', 'client'],
+    ['custom1', 'custom2', 'custom3', 'custom4', 'custom5',
+      'custom6', 'custom7', 'custom8', 'custom9', 'custom10']
+  );
+  
+  const customFields = (function() {
+    let params = '';
+    for (let i = 1; i <= 10; i++) {
+      const fieldName = 'custom' + i;
+      if (data[fieldName]) {
```

Comment:
Would leave to you to decide whether this is intended. If we set `custom1=0`, this line would return false and the value will not be included in the arguments.

Reply:
That's a good point.  I've corrected it to check for undefined instead.